### ISSUE TITLE
[YSQL] auto-analyze: skip analyze calls for secondary indexes

### ISSUE THREAD

**Issue by yifanguan** [yugabyte/yugabyte-db#23506]
Jira Link: [DB-12428](https://yugabyte.atlassian.net/browse/DB-12428)
### Description

We should only analyze tables in auto analyze service.
Currently, we aggregate mutation counts and call analyze for indexes as well.
Any secondary index is analyzed along with its base table when its base table analyze is run.

### Issue Type

kind/enhancement

### Warning: Please confirm that this issue does not contain any sensitive information

- [X] I confirm this issue does not contain any sensitive information.

[DB-12428]: https://yugabyte.atlassian.net/browse/DB-12428?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by yifanguan** [comment]
PG cannot analyze non-tables or special system tables. We should follow the same pattern.
```
postgres=# analyze pg_class_relname_nsp_index;
WARNING:  skipping "pg_class_relname_nsp_index" --- cannot analyze non-tables or special system tables
ANALYZE
```

**Comment by yifanguan** [comment]
issue is resolved by commit ee639f4
